pusher/crypto.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import (
+    print_function,
+    unicode_literals,
+    absolute_import,
+    division)
+
+import hashlib
+import nacl
+import base64
+
+from pusher.util import (
+    ensure_text,
+    data_to_string)
+
+import nacl.secret
+import nacl.utils
+
+# The prefix any e2e channel must have
+ENCRYPTED_PREFIX = 'private-encrypted-'
+
+def is_encrypted_channel(channel):
+    """
+    is_encrypted_channel checks() if the channel is encrypted by verifying the prefix
+    """
+    if channel.startswith(ENCRYPTED_PREFIX):
+        return True
+    return False
+
+def is_encryption_master_key_valid(encryption_master_key):
+    """
+    is_encryption_master_key_valid() checks if the provided encryption_master_key is validby checking its length
+    """
+    if len(encryption_master_key) == 32:
+        return True
+
+    return False
+
+def generate_shared_secret(channel, encryption_master_key):
+    """
+    generate_shared_secret() takes a utf8-string
+    and returns the sha256 hash in utf8-string format
+    """
+
+    if encryption_master_key is not None:
+
+        encryption_master_key = ensure_text(encryption_master_key, "encryption_master_key")
```

Comment:
Yes, we should treat everything (meaning the `encryption_master_key` and the `channel` name ) as `bytes` they are both used to construct the `shared_secret`.
This has a small caveat since bytes are not represented by the same types in Python 2.x and Python 3.x ( more here: https://pythonhosted.org/six/#six.binary_type and here: https://hacktec.gitbooks.io/effective-python/content/en/Chapter1/Item3.html ) and we should keep this in consideration for the edit.